lib/service/connect.go

```diff
@@ -447,18 +447,35 @@ func (process *TeleportProcess) periodicSyncRotationState() error {
 	}
 
 	periodic := interval.New(interval.Config{
-		Duration:      defaults.HighResPollingPeriod,
-		FirstDuration: utils.HalfJitter(defaults.HighResPollingPeriod),
+		Duration:      pollingInterval,
+		FirstDuration: utils.HalfJitter(pollingInterval),
 		Jitter:        utils.NewSeventhJitter(),
 	})
 	defer periodic.Stop()
 
+	errors := utils.NewTimedCounter(process.Clock, syncRotationFailureThreshold.Time)
+
 	for {
 		err := process.syncRotationStateCycle()
 		if err == nil {
 			return nil
 		}
-		process.log.Warningf("Sync rotation state cycle failed: %v, going to retry after ~%v.", err, defaults.HighResPollingPeriod)
+		process.log.WithError(err).Warningf("Sync rotation state cycle failed: %v", err)
+
+		// If we have had a *lot* of failures very recently, then it's likely that our
+		// route to the auth server is gone. If we're using a tunnel then it's possible
+		// that the proxy has been reconfigured and the tunnel address has moved.
+		count := errors.Increment()
+		process.log.Warnf("%d connection errors in last %v.", count, syncRotationFailureThreshold.Time)
+		if count > syncRotationFailureThreshold.Amount {
+			// signal quit
+			process.log.Error("Connection error threshold exceeded. Asking for a graceful restart.")
+			process.BroadcastEvent(Event{Name: TeleportReloadEvent})
+			return nil
+		}
+
+		process.log.Warningf("Retrying in ~%v", defaults.HighResPollingPeriod)
```

Comment:
Suggestion: Ditch this log, or merge it with the one at line 463? One warning for the problem should be enough.

If you decide keep it, please change `defaults.HighResPollingPeriod` to `pollingInterval`.

Reply:
Kept, but changed logged var to `pollingInterval`.